pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -527,6 +528,12 @@ func (s *FlinkStateMachine) handleApplicationDeleting(ctx context.Context, app *
 		return err
 	}
 
+	if len(jobs) == 0 {
```

Comment:
Shouldn't this be `if finished ` ? -  based on the comment below ?

Reply:
No, we already handle that case in jobFinished. This handles the case that the job finished or was canceled and the jobmanager restarted, in which case the job will no longer be returned from the API.

Reply:
@mwylde Check this issue - https://github.com/lyft/flinkk8soperator/issues/13

This is scenario - DeleteMode = DeleteModeSavepoint, the underlying job is finished or cancelled. The deletion is stuck.

I hit this case when migrating to controller runtime, where we failed to store the triggerId in etcd. In the next loop, the job was already cancelled.

Reply:
Sorry, you're totally right. That case is now handled correctly (with a test).